code/game/visual_objects.dm

```diff
@@ -19,7 +19,11 @@
 	. = ..()
 	if(new_icon_state)
 		icon_state = new_icon_state
-	add_overlay(emissive_appearance(icon, icon_state, src, layer, attached = TRUE))
+	update_icon()
```

Comment:
use update_appearance. THen if you're specifically only needing to update overlays, use the update overlays flag